tests/model_registry/conftest.py

```diff
@@ -277,11 +347,18 @@ def after_call(self, response: Response, case: Case) -> None:
 
 @pytest.fixture(scope="class")
 def updated_dsc_component_state_scope_class(
+    pytestconfig: Config,
     request: FixtureRequest,
     dsc_resource: DataScienceCluster,
     admin_client: DynamicClient,
-    is_model_registry_oauth: bool,
+    teardown_resources: bool,
 ) -> Generator[DataScienceCluster, Any, Any]:
+    if not teardown_resources or pytestconfig.option.post_upgrade:
+        # if we are not tearing down resources or we are in post upgrade, we don't need to do anything
+        # the pre_upgrade/post_upgrade fixtures will handle the rest
+        yield dsc_resource
+        return
```

Comment:
Why both yield and return?

Reply:
Because it's a Generator so we need to yield, and the execution during the teardown should do nothing and only return

Reply:
fixed with the if/else